src/lib/components/svg-view/Label.svelte

```diff
@@ -0,0 +1,38 @@
+<script lang="ts">
+	import DraggableSvg from "./DraggableSVG.svelte";
+	import type { iPoint } from "$lib/interfaces/iPoint";
+
+	export let parentPosition: iPoint;
+	export let position: iPoint;
+	export let text: string;
+</script>
+
+<DraggableSvg bind:position>
+	<!-- line from parrent to text label, make dotted-->
+	<line
+		stroke-dasharray="5,5"
+		d="M5 20 l215 0"
+		x1={parentPosition.x}
+		y1={parentPosition.y}
+		x2={position.x}
+		y2={position.y}
+		stroke="gray"
+		stroke-width="2"
+	/>
+
+	<!-- text label background -->
+	<rect
+		x={position.x - text.length * 5}
+		y={position.y - 10}
+		width={text.length * 10}
+		height="20"
+		fill="darkgray"
+		text-anchor="middle"
+		stroke-width="2"
+	/>
+
+	<!-- text label -->
+	<text x={position.x} y={position.y + 5} text-anchor="middle" role="none">
```

Comment:
What is this random `+ 5`? If it is necessary, please define it as a variable with an explanation.

Reply:
it moves the text into the rectangle, how would you suggest going about the variable just in the script?

Reply:
Yes put a const in the script